Added Absolute path operations to File

Ezbob opened the pull request:
The motivation for this enhancement is to provide operations in the File interface to convert relative file paths into absolute file paths, using the Java 7 `Path` class.

The operations does not check whether the files exists (this is a task for the user if they want to), so they can be used to create new file paths for files that does not exist.

fmontesi commented on include/file.iol:
```diff
@@ -77,6 +77,18 @@ type ListResponse:void {
 interface FileInterface {
 RequestResponse:
 	/**!
+	 * Constructs a absolute path to the target file or directory. 
```
a absolute -> an absolute

fmontesi commented on include/file.iol:
```diff
@@ -77,6 +77,18 @@ type ListResponse:void {
+	 * Constructs an absolute path to the target file or directory. 
+	 * Can be used to construct an absolute path for new files that does not exist yet.
+	 */
+	getAbsolutePath( string )( string ),
```
I'd probably call it toAbsolutePath, following Java in toString, toAbsolutePath, etc.

@jolie/developers Comments on the convention for transformer operations? I don't have *very* strong feelings here. (yet?) :-)

fmontesi commented on include/file.iol:
```diff
@@ -77,6 +77,18 @@ type ListResponse:void {
+	 * Constructs an absolute path to the parent directory of the target file or directory.
+	 * Can be used to construct an absolute path to the parent directory for files that does not exists yet.
+	 */
+	getAbsoluteParentPath( string )( string ),
```
Why not simply getParentPath ? Then using this becomes a matter of:

toAbsolutePath@File( path )( path );
getParentPath@File( path )( path )

fmontesi commented:
Awesome, thanks. I couldn't finish my comments before, put the last ones now.

Ezbob commented:
Refactored the code a again, as suggested by @fmontesi . 

This time I also added faults; if the path is not recognized as a valid path by the file system then we get a fault, and in the case of `getParentPath` if the parent path cannot exist (i.e. get the parent of the root "/" on nix machines) then we also get a fault.

fmontesi commented on javaServices/coreJavaServices/src/joliex/io/FileService.java:
```diff
@@ -744,4 +745,38 @@ public boolean accept( File directory, String filename )
+		String absolutePath = Paths.get( fileName ).toAbsolutePath().normalize().toString();
+
+		if ( absolutePath == null ) {
+			throw new FaultException( "Not a valid path", new IOException() );
```
Should be

throw new FaultException( new IOException( "Not a valid path" ) );

Same for the other method.

fmontesi commented:
Thanks! I added a comment on a bug I noticed only now, and then we're good to go!

Ezbob commented:
Ahh... We got an unchecked exception that `Paths.get` throws... I'm just gonna wrap that up in a `FaultException`..

fmontesi commented on include/file.iol:
```diff
@@ -77,6 +77,20 @@ type ListResponse:void {
+	 * Can be used to construct an absolute path for new files that does not exist yet.
+	 * Throws a fault if input is a relative path is not system recognized path.
+	 */
+	toAbsolutePath( string )( string ),
```
Should declare that it can throw InvalidPathException, with its type (probably JavaExceptionType?). Same for the one below.

fmontesi commented on include/file.iol:
```diff
@@ -77,6 +77,20 @@ type ListResponse:void {
+	 * Can be used to construct an absolute path for new files that does not exist yet.
+	 * Throws a InvalidPathException fault if input is a relative path is not system recognized path.
+	 */
+	toAbsolutePath( string )( string ) throws JavaExceptionType( InvalidPathExceptionType ),
```
Huh? How does this work?

Should probably be: `throws InvalidPathException( JavaExceptionType )`

Same below.

fmontesi commented:
Oh, sorry, didn't notice that this was updated. I put in a last comment, promise. :-) I think that your interface would not compile now.

Ezbob commented:
Should be fixed now.

fmontesi commented:
Thanks. Merged. :+1:
